lib/OpenQA/WebAPI/Plugin/ObsRsync.pm

```diff
@@ -333,4 +344,32 @@ sub _check_error {
     return 0;
 }
 
+sub _for_every_batch {
+    my ($c, $project, $sub) = @_;
+    my $helper = $c->obs_rsync;
+    my ($parent_project, $thisbatch) = $helper->split_project_batch($project);
+
+    # Three cases possible when $sub() must be called:
+    # for this single batch alone ($thisbatch != undef)
+    # for a regular project without batches (!@$batches)
+    # for every batch of a project with batches
+    if ($thisbatch) {
+        return &$sub($parent_project, $thisbatch);
```

Comment:
I'm wondering about this syntax. I assume `$sub` is a coderef which is supposed to be called at this point. Isn't `$sub->(...)` the more commonly used syntax for this? (@kraih)

Reply:
Yes, `$sub->(...)` is the correct form.